lib/signalfx.js

```diff
@@ -0,0 +1,334 @@
+  var name = 'signalfx-nodejs-client';
+  var version = '0.1.0';
+
+// Default Parameters
+  var DEFAULT_INGEST_ENDPOINT = 'https://ingest.signalfx.com';
+  var DEFAULT_API_ENDPOINT = 'https://api.signalfx.com';
+  var DEFAULT_BATCH_SIZE = 300;// Will wait for this many requests before posting
+  var DEFAULT_TIMEOUT = 1500; // Default timeout is 1s
+
+// Global Parameters
+  var PROTOBUF_HEADER_CONTENT_TYPE = 'application/x-protobuf';
+  var JSON_HEADER_CONTENT_TYPE = 'application/json';
+
+
+// ============================================================================================
+//                            Check if we can use Protobuf
+// ============================================================================================
+  var useProtocolBuffers = true;
+  var protocolBuffers;
+  try {
+    protocolBuffers = require('./signal_fx_protocol_buffers_pb2').com.signalfx.metrics.protobuf;
+  } catch (error) {
+    winston.warn('Protocol Buffers not installed properly. Switching to Json.', error);
+    useProtocolBuffers = false;
+  }
+
+// ============================================================================================
+//                            BaseSignalFx Signal Fx class
+// ============================================================================================
+  function BaseSignalFx(apiToken, ingestEndpoint,
+                        apiEndpoint, timeout,
+                        batchSize, userAgents) {
+    this.apiToken = apiToken;
+    this.ingestEndpoint = ingestEndpoint || DEFAULT_INGEST_ENDPOINT;
+    this.apiEndpoint = apiEndpoint || DEFAULT_API_ENDPOINT;
+    this.timeout = timeout || DEFAULT_TIMEOUT;
+    this.batchSize = Math.max(1, (batchSize ? batchSize : DEFAULT_BATCH_SIZE));
+    this.userAgents = userAgents || null;
+  }
+
+  BaseSignalFx.prototype.send = function (cumulativeCounters, gauges, counters) {
+    if (!cumulativeCounters && !gauges && !counters) {
+      return;
+    }
+
+    return {
+      cumulative_counter: cumulativeCounters,
+      gauge: gauges,
+      counter: counters
+    };
+  };
+
+  BaseSignalFx.prototype.sendEvent = function (eventType, dimensions, properties) {
+    return {
+      eventType: eventType,
+      dimensions: dimensions || [],
+      properties: properties || []
+    };
+  };
+
+
+// ============================================================================================
+//                          Client Signal Fx class - base API client
+// ============================================================================================
+  /**
+   * SignalFx API client.
+   * This class presents a programmatic interface to SignalFx's metadata and
+   * ingest APIs. At the time being, only ingest is supported; more will come
+   * later.
+   * @param apiToken
+   * @constructor
+   */
+  function SignalFxClient(apiToken) {
+    BaseSignalFx.apply(this, arguments);
+
+    this.HEADER_API_TOKEN_KEY = 'X-SF-Token';
```

Comment:
Generally you can apply constants such as these directly to the prototype to keep the constructor free of clutter. For example:

  SignalFxClient.prototype.HEADER_API_TOKEN_KEY = 'X-SF-Token';

(Keep in mind you can't do this for object instantiations such as this.queue = []; since they need to be assigned to the object instance rather than the prototype)